kge/model/kge_model.py

```diff
@@ -426,11 +458,62 @@ def create_from(
             if not config.log_folder or not os.path.exists(config.log_folder):
                 config.log_folder = "."
         dataset = Dataset.create_from(checkpoint, config, dataset, preload_data=False)
+
+        # do not load pretrained embeddings on creation from checkpoint
+        entity_pretrain_filename = config.get_default(
+            config.get("model") + ".entity_embedder.pretrain.model_filename"
+        )
+        relation_pretrain_filename = config.get_default(
+            config.get("model") + ".relation_embedder.pretrain.model_filename"
+        )
+        config.set(config.get("model") + ".entity_embedder.pretrain.model_filename", "")
+        config.set(
+            config.get("model") + ".relation_embedder.pretrain.model_filename", ""
+        )
+
         model = KgeModel.create(config, dataset)
+        config.set(
+            config.get("model") + ".entity_embedder.pretrain.model_filename",
+            entity_pretrain_filename,
+        )
+        config.set(
+            config.get("model") + ".relation_embedder.pretrain.model_filename",
+            relation_pretrain_filename,
+        )
         model.load(checkpoint["model"])
         model.eval()
         return model
 
+    def init_entities_pretrained(self, pretrained_model: "KgeModel", ensure_all=False):
+        """
+        Initializes entity embedder with pre-trained embeddings from a packaged model
+        Args:
+            pretrained_embedder: KgeModel containing pre-trained entity embeddings
+            ensure_all: ensure that all entities can be initialized with
+                        embeddings provided in the pre-trained model
+        """
+        self._entity_embedder.init_pretrained(
+            pretrained_model._entity_embedder._embeddings,
+            pretrained_model.dataset.entity_ids(),
+            self.dataset.entity_ids(),
+            ensure_all=ensure_all,
+        )
+
+    def init_relations_pretrained(self, pretrained_model: "KgeModel", ensure_all=False):
+        """
+        Initializes relation embedder with pre-trained embeddings from a packaged model
+        Args:
+            pretrained_embedder: KgeModel containing pre-trained relation embeddings
+            ensure_all: ensure that all relations can be initialized with
+                        embeddings provided in the pre-trained model
+        """
+        self._relation_embedder.init_pretrained(
+            pretrained_model._relation_embedder._embeddings,
+            pretrained_model.dataset.relation_ids(),
+            self.dataset.relation_ids(),
+            ensure_all=ensure_all,
+        )
+
```

Comment:
These two methods are not needed anymore once we take the ids of the embedder's dataset directly.